src/presentation/api/controller/services.go

```diff
@@ -0,0 +1,212 @@
+	return &ServicesController{
+		servicesService: service.NewServicesService(persistentDbService),
+	}
+}
+
+// ReadServices	 godoc
+// @Summary      ReadServices
+// @Description  List installed services and their status.
+// @Tags         services
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} dto.InstalledServiceWithMetrics
+// @Router       /v1/services/ [get]
+func (controller *ServicesController) Read(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(c, controller.servicesService.Read())
+}
+
+// ReadInstallableServices	 godoc
+// @Summary      ReadInstallableServices
+// @Description  List installable services.
+// @Tags         services
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} entity.InstallableService
+// @Router       /v1/services/installables/ [get]
+func (controller *ServicesController) ReadInstallables(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(
+		c, controller.servicesService.ReadInstallables(),
+	)
+}
+
+func parseRawPortBindings(bindings []interface{}) []string {
+	rawPortBindings := []string{}
+	for _, rawPortBinding := range bindings {
+		rawPortBindingMap, assertOk := rawPortBinding.(map[string]interface{})
+		if !assertOk {
+			slog.Debug(
+				"InvalidPortBindingStructure", slog.Any("portBinding", rawPortBinding),
+			)
+			continue
+		}
+
+		rawPortStr, err := voHelper.InterfaceToString(rawPortBindingMap["port"])
+		if err != nil {
+			slog.Debug(err.Error(), slog.String("port", rawPortStr))
+			continue
+		}
+
+		rawProtocolStr, err := voHelper.InterfaceToString(rawPortBindingMap["protocol"])
+		if err != nil {
+			slog.Debug(err.Error(), slog.String("port", rawPortStr))
+			continue
+		}
+
+		rawPortBindingStr := rawPortStr + "/" + rawProtocolStr
+		rawPortBindings = append(rawPortBindings, rawPortBindingStr)
+	}
+
+	return rawPortBindings
+}
+
+// CreateInstallableService godoc
+// @Summary      CreateInstallableService
+// @Description  Install a new installable service.
+// @Tags         services
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        createInstallableServiceDto	body dto.CreateInstallableService	true	"Only name is required.<br />If version is not provided, it will be 'lts'.<br />If portBindings is not provided, it wil be default service port bindings.<br />If autoCreateMapping is not provided, it will be 'true'."
+// @Success      201 {object} object{} "InstallableServiceCreated"
+// @Router       /v1/services/installables/ [post]
+func (controller *ServicesController) CreateInstallable(c echo.Context) error {
+	requestBody, err := apiHelper.ReadRequestBody(c)
+	if err != nil {
+		return err
+	}
+
+	rawEnvsSlice := []string{}
+	if requestBody["envs"] != nil {
+		for _, rawEnv := range requestBody["envs"].([]interface{}) {
+			rawEnvStr, err := voHelper.InterfaceToString(rawEnv)
+			if err != nil {
+				continue
```

Comment:
Like I mentioned before, everytime you discard something you should put a slog.Debug, specially on for loops.

Reply:
Weird... I remember adding it in all loop cases, including the one mentioned. Well, I'm adding it again anyway.